toolkit/tools/imagegen/configuration/configuration.go

```diff
@@ -93,6 +95,27 @@ func (c *Config) GetDiskContainingPartition(partition *Partition) (disk *Disk) {
 	return nil
 }
 
+// GetKernelCmdLineValue returns the output of a specific option setting in /proc/cmdline
+func GetKernelCmdLineValue(option string, startIndex int) (cmdlineValue string, err error) {
```

Comment:
>startIndex [](http://example.com/codeflow?start=42&length=10)

Is it possible to calculate `startIndex` from `option`? From the use of this function it seems like it points to the first character after `=`. Would `len(option) + 1` work?

Reply:
No, there're options that do not go with this format. Have modified the caller comment to better explain

Reply:
Could we design out the part where the caller needs to inform the callee of the starting index? It really seems like an easy pitfall for the caller to mess up.

Something like condition on first `=` or error if you get to end of parameter (instance of ` ` or end of line"

Reply:
I have redesigned this by eliminating the starting index. The rest of the string processing will be left to the specific caller to perform.